classes/PodsAPI.php

```diff
@@ -701,7 +701,23 @@ public function save_term( $term_data, $term_meta, $strict = false, $sanitized =
 
 			unset( $term_data['name'] );
 
-			$term_data['term_id'] = wp_insert_term( $term_name, $taxonomy, $term_data );
+			// Check to be certain the 'new' term doesn't exist.
+			if ( is_int( $term_name ) ) {
+				$term = get_term_by( 'term_id', $term_name, $taxonomy );
+			} else {
+				$term = get_term_by( 'name', $term_name, $taxonomy );
+
+				if ( ! $term ) {
+					$term = get_term_by( 'slug', $term_name, $taxonomy );
```

Comment:
Is this necessary for your use-case?

Reply:
@mgratch

Reply:
So after reviewing this further, this PR seems unnecessary. Core adds terms by the same process and doesn't do any last minute checks.  Validation should happen before this, in my case in another plugin.